src/esphome/sensor/mhz19_component.cpp

```diff
@@ -58,6 +100,19 @@ void MHZ19Component::update() {
 
 bool MHZ19Component::mhz19_write_command_(const uint8_t *command, uint8_t *response) {
   this->flush();
+
+  /*
+   * UART0 in NodeMCU v2, sometimes on boot has junk in RX buffer,
+   * check for it and drain all of it before sending commands to sensor
```

Comment:
Uhm the `flush` command should flush the RX buffer (naming is bad, but copied from arduino core). If it does not do so that would be a bug.

Reply:
That's make on it as well,
but got arduino/esp only a couple days ago so I'm not sure where to look for respective code
and this workaround fixed non working sensor for me.

If it's within this git repo and you can point out to related code, I'll give it a try.